src/rgw/radosgw-admin/radosgw-admin.cc

```diff
@@ -4886,38 +4886,40 @@ int main(int argc, const char **argv)
         boost::optional<RGWRESTConn> conn;
         RGWRESTConn *remote_conn = nullptr;
         if (url.empty()) {
-          // load current period for endpoints
-          RGWRealm realm;
-          int ret = rgw::read_realm(dpp(), null_yield, cfgstore.get(),
-                                    realm_id, realm_name, realm);
-          if (ret < 0 ) {
-            cerr << "failed to load realm: " << cpp_strerror(-ret) << std::endl;
-            return -ret;
-          }
-          period_id = realm.current_period;
+          cerr << "A --url must be provided." << std::endl;
+          return EINVAL;
+        }
+        // load current period for endpoints
+        RGWRealm realm;
+        int ret = rgw::read_realm(dpp(), null_yield, cfgstore.get(),
+                                  realm_id, realm_name, realm);
+        if (ret < 0 ) {
+          cerr << "failed to load realm: " << cpp_strerror(-ret) << std::endl;
+          return -ret;
+        }
+        period_id = realm.current_period;
 
-          RGWPeriod current_period;
-          ret = cfgstore->read_period(dpp(), null_yield, period_id,
-                                      std::nullopt, current_period);
-          if (ret < 0) {
-            cerr << "failed to load current period: " << cpp_strerror(-ret) << std::endl;
-            return -ret;
-          }
-          if (remote.empty()) {
-            // use realm master zone as remote
-            remote = current_period.get_master_zone().id;
-          }
-          conn = get_remote_conn(static_cast<rgw::sal::RadosStore*>(driver), current_period.get_map(), remote);
-          if (!conn) {
-            cerr << "failed to find a zone or zonegroup for remote "
-                << remote << std::endl;
-            return -ENOENT;
-          }
-          remote_conn = &*conn;
+        RGWPeriod current_period;
+        ret = cfgstore->read_period(dpp(), null_yield, period_id,
+                                    std::nullopt, current_period);
+        if (ret < 0) {
+          cerr << "failed to load current period: " << cpp_strerror(-ret) << std::endl;
+          return -ret;
+        }
+        if (remote.empty()) {
+          // use realm master zone as remote
+          remote = current_period.get_master_zone().id;
+        }
+        conn = get_remote_conn(static_cast<rgw::sal::RadosStore*>(driver), current_period.get_map(), remote);
+        if (!conn) {
+          cerr << "failed to find a zone or zonegroup for remote "
+              << remote << std::endl;
+          return -ENOENT;
         }
+        remote_conn = &*conn;
```

Comment:
`do_period_pull()` calls `send_to_remote_or_url()`, so it requires either a url or a `RGWRESTConn*`

if we're requiring a `--url`, then none of this should be necessary - it's only used to get the `RGWRESTConn* remote_conn` which `send_to_remote_or_url()` will ignore

Reply:
ok right. I suppose we should just rely on url like other commands.